dev/release-notes.py

```diff
@@ -0,0 +1,211 @@
+    # is not really helping - could be that some very old PRs are being merged.
+    for pr in all_pulls:
+        print(pr.number, end=" ")
+        # flush stdout immediately, to see progress indicator
+        sys.stdout.flush()
+        if pr.merged:
+            if pr.closed_at > datetime.fromisoformat(startdate):
+                labs = [lab.name for lab in list(pr.get_labels())]
+                prs[pr.number] = { "title" : pr.title,
+                                    "closed_at" : pr.closed_at.isoformat(),
+                                    "labels" : labs }
+#                 if len(prs)>5: # for quick testing (maybe later have an optional argument)
+#                     break
+    print("\n")
+    with open("prscache.json", "w", encoding="utf-8") as f:
+        json.dump(prs, f, ensure_ascii=False, indent=4)
+    return prs    
+
+def changes_overview(prs,startdate):
+    """Writes files with information for release notes."""
+
+    #TODO: using cached data, check that the starting date is the same
+    # Also save the date when cache was saved (warning if old?)
+    print("## Changes since", startdate)
+
+    # Opening files with "w" resets them
+    f = open("releasenotes.md", "w")
+    f2 = open("remainingPR.md", "w")
+    f3 = open("releasenotes.json", "w")
+    jsondict = prs.copy()
+
+    prioritylist = [
+        ["kind: bug: wrong result", "Fixed bugs that could lead to incorrect results"],
+        ["kind: bug: crash", "Fixed bugs that could lead to crashes"],
+        ["kind: bug: unexpected error", "Fixed bugs that could lead to break loops"],
+        ["kind: bug", "Other fixed bugs"],
+        ["kind: enhancement", "Improved and extended functionality"],
+        ["kind: new feature", "New features"], 
+        ["kind: performance", "Performance improvements"],
+        ["topic:libgap", "Improvements in the experimental way to allow 3rd party code to link GAP as a library"],
+        ["topic: julia", "Improvements in the experimental support for using the **Julia** garbage collector"],
+        ["topic: documentation", "Changed documentation"],
+        ["topic: packages", "Packages"]
+    ]
+
+    # TODO: why does this need a special treatment? 
+    # Adding it to the prioritylist could ensure that it goes first
+    f.write("## Release Notes \n\n")
+    f.write("### " + "New features and major changes" + "\n")
+    removelist = []
+    for k in prs:
+        # The format of an entry of list is: ["title of PR", "Link" (Alternative the PR number can be used), [ list of labels ] ]
+        if "release notes: highlight" in prs[k]["labels"]:
+            # TODO: writing up these details should be a function
+            f.write("- [#")
+            issuenumber = str(k)
+            f.write(issuenumber)
+            f.write("](")
+            f.write("https://github.com/gap-system/gap/pull/" + str(k))
+            f.write(") ")
+            f.write(prs[k]["title"])
+            f.write("\n")
```

Comment:
This strikes me as rather unpythonic code... How about this instead, using an f-string (and assuming that `k` is an integer -- is it?):
```suggestion
            title = prs[k]["title"]
            f.write(f"- [#{k}](https://github.com/gap-system/gap/pull/{k}) {title}\n")
```